api/document/serializers/doc_cursor.py

```diff
@@ -118,15 +118,55 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.context['parent_serializer'] = self
 
+    @property
+    def is_root(self) -> bool:
+        return self.context.get('is_root', True)
+
     def get_meta(self, instance):
         """Include meta data which applies to the whole node."""
         meta = Meta(
             instance,
-            self.context.get('is_root', True),
+            self.is_root,
             self.context.get('policy'),
         )
         return MetaSerializer(meta, context={'parent_serializer': self}).data
 
+    def validate_footnote_citations(self, data: PrimitiveDict,
+                                    footnote_emblems: Set[str]):
+        citations = [
+            util.get_content_text(f['inlines']).strip()
+            for f in util.iter_inlines(data['content'])
+            if f['content_type'] == 'footnote_citation'
+        ]
+
+        for citation in citations:
+            if citation not in footnote_emblems:
+                raise ValidationError(
+                    f"Citation for '{citation}' has no matching footnote"
+                )
+
+    def validate_footnote_emblems(self, data: PrimitiveDict) -> Set[str]:
+        footnote_emblems = [
+            f['type_emblem'] for f in util.iter_children(data['children'])
+            if f['node_type'] == 'footnote'
+        ]
```

Comment:
Similarly, this feels like a validation that should occur _after_ creating the tree. What do you think?